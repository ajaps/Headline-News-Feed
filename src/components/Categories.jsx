import React from 'react';
import PropTypes from 'prop-types';
import Category from '../components/Category.jsx';

/**
 * Represents a navigation bar containing different categories
 * @param {props} props react props
 * @return {React} returns react componenet
 */
const Categories = (props) => {
  const categoryObj = {};
  const currentCategory = props.currentcategory[0];
  const allSources = props.allCategory;
  const filterCategoryInSource = allSources.map((categoryItem) => {
    if (!(categoryItem.category in categoryObj)) {
      categoryObj[categoryItem.category] = categoryItem.category;
      return (
        <Category
            category={categoryItem}
            currentCategory={currentCategory}
            key={categoryItem.category}
        />);
    }
  });

  return (
    <nav className="navbar navbar-default categoryList">
      <div className="container-fluid">
        <div className="navbar-header">
          <h5 className="text-muted hidden">News Category</h5>
          <button aria-controls="navbar" aria-expanded="false"
              className="navbar-toggle collapsed" data-target="#navbar"
              data-toggle="collapse" type="button"
          >
            <span className="sr-only">Toggle navigation</span>
            <span className="icon-bar" />
            <span className="icon-bar" />
            <span className="icon-bar" />
          </button>
        </div>
        <div className="navbar-collapse collapse" id="navbar" >
          <ul className="nav nav-justified">
            { filterCategoryInSource }
          </ul>
        </div>
      </div>
    </nav>
  );
};

Categories.propTypes = {
  allCategory: PropTypes.array,
  currentcategory: PropTypes.array,
};

export default Categories;
