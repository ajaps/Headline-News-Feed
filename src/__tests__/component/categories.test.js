import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Categories from '../../components/Categories.jsx';

jest.mock('../../api/NewsApi', () => ({
  getData: () => Promise.resolve('getPromise')
}));

describe('Category component', () => {
  const sources = [{ category: 'general' }, { category: 'sport' }];
  let app;
  beforeEach(() => {
    app = shallow(
      <Categories allCategory={sources}
          currentcategory={['general']}
      />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });
});
