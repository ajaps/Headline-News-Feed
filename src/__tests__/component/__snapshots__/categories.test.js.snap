// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component should render as expected 1`] = `
<nav
  className="navbar navbar-default categoryList"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <h5
        className="text-muted hidden"
      >
        News Category
      </h5>
      <button
        aria-controls="navbar"
        aria-expanded="false"
        className="navbar-toggle collapsed"
        data-target="#navbar"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      className="navbar-collapse collapse"
      id="navbar"
    >
      <ul
        className="nav nav-justified"
      >
        <Category
          category={
            Object {
              "category": "general",
            }
          }
          currentCategory="general"
        />
        <Category
          category={
            Object {
              "category": "sport",
            }
          }
          currentCategory="general"
        />
      </ul>
    </div>
  </div>
</nav>
`;
